src/databricks/labs/remorph/reconcile/query_builder/sampling_query.py

```diff
@@ -0,0 +1,57 @@
+from pyspark.sql import DataFrame
+
+from databricks.labs.remorph.reconcile.constants import SourceType
+from databricks.labs.remorph.reconcile.query_builder.base import QueryBuilder
+from databricks.labs.remorph.reconcile.recon_config import TransformRuleMapping
+
+
+class SamplingQueryBuilder(QueryBuilder):
+
+    def build_query(self, df: DataFrame):
+        key_cols = sorted(self.table_conf.join_columns)
+        keys_df = df.select(*key_cols)
+        with_clause = self._get_with_clause(keys_df, self.source)
+
+        cols = sorted(
+            (self.table_conf.join_columns | self.select_columns)
+            - self.table_conf.threshold_columns
+            - self.table_conf.drop_columns
+        )
+
+        col_transform = self._generate_transform_rule_mapping(cols)
+        cols_expr = sorted(self._get_column_expr(TransformRuleMapping.get_column_expr_with_alias, col_transform))
+
+        query_filter = self.table_conf.get_filter(self.layer)
+        select_query = self._construct_query(self.table_name, query_filter, cols_expr, with_clause, key_cols)
+        return select_query
+
+    @staticmethod
+    def _construct_query(
+        table: str, query_filter: str, col_expr: list[str], with_clause: str, key_cols: list[str]
+    ) -> str:
+
+        with_cte = f"{with_clause} src as ( select " + ",".join(col_expr) + f" from {table} where {query_filter} )"
+
+        join_expr = [f"recon.{col} = src.{col}" for col in key_cols]
+        join_condition = " and ".join(join_expr)
+        join = f" select src.* from src join recon on {join_condition}"
+
+        select_query = with_cte + join
+        return select_query
+
+    @staticmethod
+    def _get_with_clause(df: DataFrame, source: str):
+        select_clause = []
+        for row in df.take(50):
```

Comment:
`50` looks like a magic number here. Also, `sample` might be better suited than `take`. 

On a more general standpoint, my gut feeling is that this function's output would be easier to format as:
`values ( x0, y0, ...), (x1, y1, ...), (xn, yn, ...) as <some alias> (column0, column1, ...)`. It would also probably avoid the need for a special case for the Oracle dialect.

Reply:
expressed as sqlglot expressions will as we can generate SQL in any target dialect

Reply:
> my gut feeling is that this funct

For sample ,we always want to get minimal sample records in smaller, medium and large volume tables say for now 50 records.If we use sample we need to keep fraction as dynamic and we need total count which will add few more extra work to worker nodes.
Table Values constructor fails in oracle with invalid table name.

Reply:
Also, `50` is still a magic number, please define it as a named constant